addons/mail/static/src/core/common/message_actions.js

```diff
@@ -135,7 +135,8 @@ messageActionsRegistry
     })
     .add("download_files", {
         condition: (component) =>
-            component.message.attachment_ids.length > 1 && component.store.self.isInternalUser,
+            component.message.attachment_ids.length > 1 &&
+            component.store.self.main_user_id?.share === false,
```

Comment:
Not convinced by the `=== false` either, but it's more acceptable than for truthy check.